ambry-store/src/test/java/com.github.ambry.store/StatsManagerTest.java

```diff
@@ -0,0 +1,270 @@
+    preAggregatedSnapshot = generateStatsSnapshot(testData1);
+    int[][] testData2 = decomposeArray(testData1);
+    int[][] testData3 = decomposeArray(testData1);
+    storeMap = new HashMap<>();
+    statsSnapshots = new ArrayList<>();
+    statsSnapshots.add(generateStatsSnapshot(testData1));
+    statsSnapshots.add(generateStatsSnapshot(testData2));
+    statsSnapshots.add(generateStatsSnapshot(testData3));
+    for (int i = 0; i < 3; i++) {
+      PartitionId id = getTestPartitionId(i);
+      Store store = new MockBlobStore(new MockBlobStoreStats(statsSnapshots.get(i), false));
+      storeMap.put(id, store);
+    }
+    StorageManager storageManager = new MockStorageManager(storeMap);
+    Properties properties = new Properties();
+    properties.put("stats.output.file.path", outputFileString);
+    config = new StatsManagerConfig(new VerifiableProperties(properties));
+    statsManager = new StatsManager(storageManager, new ArrayList<>(storeMap.keySet()), new MetricRegistry(), config,
+        new MockTime());
+  }
+
+  /**
+   * Test to verify that the {@link StatsManager} is collecting, aggregating and publishing correctly using predefined
+   * data sets and mocked {@link Store}s and {@link StorageManager}.
+   * @throws StoreException
+   * @throws IOException
+   * @throws InterruptedException
+   */
+  @Test
+  public void testStatsManagerCollectAggregateAndPublish() throws IOException {
+    StatsSnapshot actualSnapshot = new StatsSnapshot(0L, null);
+    List<String> unreachableStores = new ArrayList<>();
+    for (PartitionId partitionId : storeMap.keySet()) {
+      statsManager.collectAndAggregate(actualSnapshot, partitionId, unreachableStores);
+    }
+    assertTrue("Actual aggregated StatsSnapshot does not match with expected snapshot",
+        preAggregatedSnapshot.equals(actualSnapshot));
+    assertEquals("Unreachable store count mismatch with expected value", 0, unreachableStores.size());
+    StatsHeader statsHeader =
+        new StatsHeader(Description.QUOTA, SystemTime.getInstance().milliseconds(), storeMap.keySet().size(),
+            storeMap.keySet().size() - unreachableStores.size(), unreachableStores);
+    File outputFile = new File(outputFileString);
+    if (outputFile.exists()) {
+      outputFile.createNewFile();
+    }
+    long fileLengthBefore = outputFile.length();
+    statsManager.publish(new StatsWrapper(statsHeader, actualSnapshot));
+    assertTrue("Failed to publish stats to file", outputFile.length() > fileLengthBefore);
+  }
+
+  /**
+   * Test to verify the behavior when {@link Store} is null and when a {@link StoreException} is thrown by the
+   * {@link StoreStats}.
+   * @throws StoreException
+   * @throws IOException
+   */
+  @Test
+  public void testStatsManagerWithProblematicStores() throws StoreException, IOException {
+    Map<PartitionId, Store> problematicStoreMap = new HashMap<>();
```

Comment:
you should also have a test where in, some stores are healthy and some are bad. And the expectation is that only the aggregated value is from those healthy ones